src/mappings/entities/mapPortfolio.ts

```diff
@@ -39,6 +40,30 @@ export const createPortfolio = (
   }).save();
 };
 
+/**
+ * Creates a Portfolio if not present.
+ * This can be need in case where someone creates an Instruction before the recipient Portfolio is created
+ */
+export const createPortfolioIfNotExists = async (
+  { identityId, number }: Pick<Portfolio, 'identityId' | 'number'>,
+  blockId: string,
+  event: SubstrateEvent
+): Promise<void> => {
+  await createIdentityIfNotExists(identityId, blockId, event);
+
+  if (number) {
```

Comment:
Don't we need to check if it exists first?

Reply:
Added the check